lib/sdr_client/update.rb

```diff
@@ -2,23 +2,62 @@
 
 module SdrClient
   # The namespace for the "update" command
-  module Update
+  class Update
     # @return [String] job id for the background job result
-    def self.run(druid, apo:, url:, logger: Logger.new($stdout))
-      cocina_item = Cocina::Models.build(
+    def self.run(druid, **options)
+      new(druid, **options).run
+    end
+
+    def initialize(druid, **options)
+      @druid = druid
+      @url = options.fetch(:url)
+      @options = options
+    end
+
+    # @return [String] job id for the background job result
+    def run
+      SdrClient::Deposit::UpdateResource.run(
+        metadata: updated_cocina_item,
+        logger: options[:logger] || Logger.new($stdout),
+        connection: Connection.new(url: url)
+      )
+    end
+
+    private
+
+    attr_reader :druid, :logger, :options, :url
+
+    def updated_cocina_item
+      @updated_cocina_item ||=
+        original_cocina_item.then { |cocina_item| maybe_update_apo(cocina_item) }
+                            .then { |cocina_item| maybe_update_collection(cocina_item) }
+    end
+
+    def original_cocina_item
+      Cocina::Models.build(
         JSON.parse(
           SdrClient::Find.run(druid, url: url)
         )
       )
+    end
 
-      SdrClient::Deposit::UpdateResource.run(
-        metadata: cocina_item.new(
-          administrative: cocina_item.administrative.new(
-            hasAdminPolicy: apo
-          )
-        ),
-        logger: logger,
-        connection: Connection.new(url: url)
+    def maybe_update_apo(cocina_item)
```

Comment:
don't love these names, but don't love `update_apo_if_needed` either ...

Maybe a method comment?

Reply:
or better - `update_apo(cocina_item) if options[:apo]` in line 32 (and ok to include guard class in method as well, but rename to `update_apo` and indicate in comments that guard clause is to avoid unintended updates)

Reply:
@ndushay Happy to rename `maybe` to `if_needed`.

I'd prefer to keep the checks in the helper methods rather than in the chain of cocina operations up above---an `if` within the `.then` block will cause a `nil` to be returned and will break the next link in the chain.

I'll rename now. Thanks!

Reply:
@ndushay ok, I think I grok the change you want. Making so you can take another look!